### PR TITLE
Firebase security

### PR DESCRIPTION
**Pull request by vitorvicente**
## Description

Finalized the transition to Firebase rules from Firebase Functions. Added & changed the Firestore Rules, deleted unused Functions, and changed Codebase to directly access the Firestore Database.

## Related Issues

https://github.com/Bostonhacks/cacioepepe/issues/102

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*

### CODE REVIEW

**Comment by rishabnayak** [comment]
please resolve conflicts!

**Comment by vitorvicente** [comment]
@rishabnayak Conflicts solved as of right now (10:45PM)

**Comment by vitorvicente** [comment]
All done, merging conflicts fixed & production bugs fixed

**Review by rishabnayak — DISMISSED** [review]

**Comment by rishabnayak** [comment]
@vitorvicente we shouldn't install the firebase admin package on the front end either because it won't be able to authenticate. to use array union, use it like this - ```firebase.firestore.FieldValue.arrayUnion()```

additionally, the function deploy script in the readme is actually correct - we have a custom deploy script because we kept hitting deploy limits!

File: README.md [hunk 1]
```diff
@@ -71,7 +71,7 @@ firebase deploy
 To deploy only functions,
 
 ```bash
-npm run deploy
+firebase deploy --only functions
```
Comment (rishabnayak) [inline README.md:74]:
```suggestion
npm run deploy
```

this actually runs a custom deploy script we wrote!

File: package.json [hunk 2]
```diff
@@ -14,6 +14,7 @@
     "chart.js": "^2.9.3",
     "core-js": "^3.6.5",
     "firebase": "^7.18.0",
+    "firebase-admin": "^9.2.0",
```
Comment (rishabnayak) [inline package.json:17]: undo

**Review by rishabnayak — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (vitorvicente) [reply package.json]: Resolved

**Comment by vitorvicente** [comment]
Removed firebase-admin dependency (sorry I didn't see it while I was revising the PR), made arrayUnion and export of /firebase/init.js

**Review by rishabnayak — APPROVED** [review]

**Comment by rishabnayak** [comment]
LGTM! Thank you so much for getting this done @vitorvicente @jy-clairechoi!

**Pull request merged**